### ISSUE TITLE
False positive undefined-variable when a function attribute of same name is set

### ISSUE THREAD

**Issue by Lucas-C** [PyCQA/pylint#3877]
### Steps to reproduce
```python
CONDITION = None
FUNC = lambda: CONDITION
FUNC.CONDITION = 42
```
Pylint output:
```
pylint_bug_repro.py:2:15: E0602: Undefined variable 'CONDITION' (undefined-variable)
```

### Current behavior
An "undefined variable" is incorrectly reported.

### Expected behavior
Zero issue detected when defining a [function attribute](https://www.python.org/dev/peps/pep-0232/)
with the same name of a local variable in the current scope.

### pylint --version output
```
$ pylint --version
pylint 2.6.0
astroid 2.4.2
Python 3.7.7 (default, Apr 10 2020, 07:59:19)
[GCC 9.3.0]
```

### Additional notes
There are a few other existing issues related to "false positive undefined-variable" errors:
- https://github.com/PyCQA/pylint/issues/2491
- https://github.com/PyCQA/pylint/issues/3688
- https://github.com/PyCQA/pylint/issues/3791

**Comment by hippo91** [comment]
@Lucas-C thanks for this clear report!

**Comment by Lucas-C** [comment]
Thanks for the fix @Pierre-Sassoulas !

**Comment by Pierre-Sassoulas** [comment]
I just reviewed ; @DanielNoord did the actual work :wink:
